Spec/Auth.swift

```diff
@@ -1885,5 +1894,104 @@ class Auth : QuickSpec {
+                defer { realtime.close() }
+                
+                // wait for connected state
+                waitUntil(timeout: testTimeout) { done in
+                    realtime.connection.once(.Connected) { stateChange in
+                        expect(stateChange!.reason).to(beNil())
+                        expect(stateChange?.current).to(equal(realtime.connection.state))
+                        done()
+                    }
+                    realtime.connect()
+                }
+                
+                // create a `rightchannel` channel and check can't attach to it
+                let channel = realtime.channels.get("rightchannel")
+                
+                waitUntil(timeout: testTimeout) { done in
+                    channel.attach() { error in
+                        expect(error).toNot(beNil())
+                        expect(error!.code).to(equal(40160))
+                        done()
+                    }
+                }
+                
+                // get second token
+                let secondTokenParams = ARTTokenParams()
+                secondTokenParams.capability = "{\"wrongchannel\": [\"*\"], \"rightchannel\": [\"*\"]}"
+                secondTokenParams.clientId = "testClientId"
+                
+                var secondToken = ""
+                var secondTokenDetails: ARTTokenDetails?
+                
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.requestToken(secondTokenParams, withOptions: nil) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails).toNot(beNil())
+                        
+                        secondToken = tokenDetails!.token
+                        secondTokenDetails = tokenDetails
+                        done()
+                    }
+                }
+                expect(secondToken).toNot(beNil())
+                expect(secondToken.characters.count > 0).to(beTrue())
+                expect(secondToken).toNot(equal(firstToken))
+                
+                // reauthorise
+                let reauthOptions = ARTAuthOptions();
+                reauthOptions.tokenDetails = secondTokenDetails
+                reauthOptions.force = true
+                
+                waitUntil(timeout: testTimeout) { done in
+                    realtime.auth.authorise(nil, options: reauthOptions) { reauthTokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(reauthTokenDetails?.token).toNot(beNil())
+                        done()
+                    }
+                }
+                
+                // re-attach to the channel
+                waitUntil(timeout: testTimeout) { done in
+                    channel.attach() { error in
+                        expect(error).toNot(beNil())
+                        expect(error!.code).to(equal(40160))
```

Comment:
I will move this test into separate PR, and it should cover `RTC8` item I guess. 
 
I took spec diffs here https://github.com/ably/docs/pull/113/files and as from my understanding - the `reauth` task contains set of spec items (RSA8e, RSA9h, RSA10k, RSA10j, RSA10d, RSA10g, RTC8, TK2d), so to implement current `reauth` task entirely - I need to implement each item, right?